src/client/src/orangehrmRecruitmentPlugin/pages/ApplyJobVacancy.vue

```diff
@@ -0,0 +1,319 @@
+import {ref} from 'vue';
+import FullNameInput from '@/orangehrmPimPlugin/components/FullNameInput';
+import SuccessDialog from '@/orangehrmRecruitmentPlugin/components/SuccessDialog';
+import {
+  maxFileSize,
+  required,
+  shouldNotExceedCharLength,
+  validEmailFormat,
+  validFileTypes,
+  validPhoneNumberFormat,
+} from '@ohrm/core/util/validation/rules';
+import SubmitButton from '@/core/components/buttons/SubmitButton';
+import {navigate} from '@/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import {urlFor} from '@/core/util/helper/url';
+
+const applicantModel = {
+  firstName: '',
+  middleName: '',
+  lastName: '',
+  contactNumber: '',
+  email: '',
+  consentToKeepData: false,
+  resume: null,
+};
+
+export default {
+  name: 'ApplyJobVacancy',
+  components: {
+    'submit-button': SubmitButton,
+    'full-name-input': FullNameInput,
+    'success-dialogue': SuccessDialog,
+  },
+  props: {
+    allowedFileTypes: {
+      type: Array,
+      required: true,
+    },
+    maxFileSize: {
+      type: Number,
+      required: true,
+    },
+    vacancyId: {
+      type: Number,
+      required: true,
+    },
+    success: {
+      type: Boolean,
+      default: false,
+    },
+    bannerSrc: {
+      type: String,
+      required: true,
+    },
+    token: {
+      type: String,
+      required: true,
+    },
+  },
+  setup() {
+    const defaultPic = `${window.appGlobal.baseUrl}/../images/logo.png`;
+    const applicant = ref({
+      ...applicantModel,
+    });
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/recruitment/public/vacancies',
+    );
+    return {
+      applicant,
+      http,
+      defaultPic,
+    };
+  },
+  data() {
+    return {
+      title: null,
+      successLabel: null,
+      subtitle: null,
+      viewMore: true,
+      isLoading: false,
+      vacancyName: null,
+      vacancyDescription: null,
+      rules: {
+        firstName: [required, shouldNotExceedCharLength(30)],
+        middleName: [shouldNotExceedCharLength(30)],
+        lastName: [required, shouldNotExceedCharLength(30)],
+        resume: [
+          maxFileSize(this.maxFileSize),
+          validFileTypes(this.allowedFileTypes),
+        ],
+        contactNumber: [shouldNotExceedCharLength(25), validPhoneNumberFormat],
+        email: [required, validEmailFormat, shouldNotExceedCharLength(50)],
+      },
+    };
+  },
+  computed: {
+    isViewDetails() {
+      return this.viewMore;
+    },
+    submitUrl() {
+      return urlFor('/recruitment/public/applicants');
+    },
+  },
+  beforeMount() {
+    this.http.get(this.vacancyId).then(response => {
+      const {data} = response.data;
+      this.vacancyName = data?.name;
+      this.vacancyDescription = data?.description;
+    });
+    if (this.success) {
+      this.title = 'Application Received';
+      this.subtitle = 'Your application has been submitted successfully';
+      this.successLabel = 'OK';
```

Comment:
@Super-Chama

Reply:
Don't really understand the requirement here. this modal is only used for this screen, can't we add these to component itself?